go/src/url_shortener/url_shortener.go

```diff
@@ -114,8 +123,52 @@ func (c *URLShortener) shortenHandler(w http.ResponseWriter, r *http.Request) {
 	c.incrementHandlerCounter(c.shortenRoute, true)
```

Comment:
Instead of having all methods on URLshortener, could we try to organize the code so that:
 - `URLshortener` only has handlers and `getURL/addURL`
 - move the stats-specific methods to `urlStatistics` 

Doing so, we could later add unit-tests of urlStatistics without having to deal with a full URLshortener objects, in order to keep things separate and more easily testable?